lightning/src/chain/keysinterface.rs

```diff
@@ -327,18 +337,29 @@ pub trait KeysInterface: Send + Sync {
 	/// A type which implements ChannelKeys which will be returned by get_channel_keys.
 	type ChanKeySigner : ChannelKeys;
 
-	/// Get node secret key (aka node_id or network_key)
+	/// Get node secret key (aka node_id or network_key).
+	///
+	/// This method must return the same value each time it is called.
 	fn get_node_secret(&self) -> SecretKey;
-	/// Get destination redeemScript to encumber static protocol exit points.
+	/// Get a script pubkey which we send funds to when claiming on-chain contestable outputs.
+	///
+	/// This method should return a different value each time it is called, for privacy reasons.
```

Comment:
Maybe you can add "to avoid clustering of channel ownership based on address reuse" ?

Reply:
Went with something more specific as "clustering" is a somewhat confusing.